File: solutions/ruby/raindrops/1/raindrops.rb
module Raindrops
  def self.convert(n)
    a = ""
    a += "Pling" if n % 3 == 0
    a += "Plang" if n % 5 == 0
    a += "Plong" if n % 7 == 0
    a.empty? ? n.to_s : a
  end
end

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "strconv"

func Convert(number int) string {
    dropMap := map[int]string{
        3: "Pling",
        5: "Plang",
        7: "Plong",
    }
	var res string

    for k, v := range dropMap {
        if (number % k == 0) { res += v }
    }

    if res == "" {
        return strconv.Itoa(number)
    }

    return res
}
